### PR TITLE
A few jenkinsfile tweaks

### PR DESCRIPTION
**Pull request by tfennelly**
# Description

See [JENKINS-37918](https://issues.jenkins-ci.org/browse/JENKINS-37918).

Misc tweaks trying to fix issues identified on the master branch.

https://ci.blueocean.io/job/ATH-Jenkinsfile/job/Jenkinsfile-tweaks/

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees 

- deleteDir at the start of the build
- remove sh to delete the blueocean-plugin dir as it was failing for some reason
- Better message for when build aborted due to uninitialised build params
- Support for etting plugins from the "latest" build on a branch

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -40,21 +40,16 @@ node ('docker') {
...
-        // Firefox browser (running in the selenium container) to make requests back
-        // in etc.
-        //
         // To bind in the local ~/.m2 when running in dev mode, simply add the following
         // volume binding to the "inside" container run settings (change username from "tfennelly"):
         //       -v /home/tfennelly/.m2:/home/bouser/.m2
         //
-        athImg.inside("--expose=12345 -p 12345:12345") {
+        athImg.inside {
             withEnv(["BLUEO_SELENIUM_SERVER_ADDR=${seleniumIP}"]) {
```
Comment (ydubreuil) [inline Jenkinsfile]: I would try to share the network stack instead of relying on the IP address to talk to Selenium by using `athImg.inside("--net:container=SELENIUM_CONTAINER_ID")` and by setting  `BLUEO_SELENIUM_SERVER_ADDR=localhost`.

Comment (tfennelly) [inline Jenkinsfile]: That does sound neater, but can we do it as a separate PR please? You can show me how to do it and we'll try it out.

Comment (tfennelly) [inline Jenkinsfile]: Is effectively the same thing though, right? They are already using the same stack from what I understand i.e. the [default network](https://docs.docker.com/engine/userguide/networking/#/default-networks) to which all containers connect, unless explicitly told to do otherwise. Anyway ... I do agree it looks a bit cleaner in the config.

Comment (tfennelly) [inline Jenkinsfile]: I'll try your above config Yoann ... gimme a few mins.

Comment (tfennelly) [inline Jenkinsfile]: @ydubreuil I get the following error ...

```
java.io.IOException: Failed to run image 'blueocean-ath-builder'. Error: flag provided but not defined: --net:container
```

Is there something else we need to do?

Comment (tfennelly) [inline Jenkinsfile]: Ok, maybe it should be ... `--net=container:NAME_or_ID`. I'll try that.

**Comment by michaelneale** [comment]
:bee: why not! nice work @ydubreuil !!

**Comment by ydubreuil** [comment]
:bee: !!!

File: Jenkinsfile (hunk 1 above)
Comment (ydubreuil) [inline Jenkinsfile]: What I meant by "same network stack" is the same network interface (same veth to be precise), so processes from both containers to feel like running on the same host, but only from the network POV

Comment (tfennelly) [inline Jenkinsfile]: @ydubreuil gotcha ... sorry for my confusion ... I actually twigged what you really meant after doing it. Certainly not the same things and certainly the way I'd prefer to do it. Thank you !!!
